main_view.py

```diff
@@ -305,43 +305,44 @@ def mem_next_state(self):
             self.active_state += 1
             self.show_state(self.active_state)
 
-    @pyqtSlot(list)
+    @pyqtSlot(object)
     def set_active_pids(self, data):
         self.view_checked_pids(data)
-        if self.active_pids_len > 0:
-            self.set_buttons(data=True, refc=False, highlight=False)
+        self.set_buttons(data=True if self.active_pids_len > 0 else False,
+                         refc=False,
+                         highlight=False)
 
-    @pyqtSlot(list)
+    @pyqtSlot(object)
     def set_collection_time(self, data):
-        self.iteration_time = data[0] if data else -1
-        self.total_time = data[1] if data else -1
+        self.iteration_time = data[0] if data is not None else -1
+        self.total_time = data[1] if data is not None else -1
 
-    @pyqtSlot(list)
+    @pyqtSlot(object)
     def set_device_data(self, data):
-        data_len = len(data)
-        self._ui.statusBar.showMessage(f'{str(*data[0]) if data_len > 0 else "No"} device was connected')
-        if data_len > 0:
-            self.devices_handler.switch(str(*data[0]))
+        if len(data) > 0:
+            self.devices_handler.switch(str(data[0][0]))
             self.device_interaction.set_device(self.devices_handler.get_device())
             self.set_buttons(pid=True, cgr=True)
+
+        self._ui.statusBar.showMessage(f'{data[0][0] if len(data) > 0 else "No"} device was connected')
         self.react()
 
     @pyqtSlot()
     def select_processes(self):
-        pids_dialog = SelectDialog(self.device_interaction.get_pid_list_all(),
+        pids_dialog = SelectDialog(data_list=self.device_interaction.get_pid_list_all(),
```

Comment:
Why data_list? Not a pid_list or just data?

Reply:
Well it's a param name chosen long ago, probably due to dialiog needs a list to create table-like-representation, and this list can contain almost anything anything